packages/sanes-browser-extension/src/extension/background/model/persona/index.ts

```diff
@@ -294,6 +302,24 @@ export class Persona {
     return filtered;
   }
 
+  public async getBalances(): Promise<readonly (readonly Amount[])[]> {
+    const accountsInfo = await this.accountManager.accounts();
+    const identities = accountsInfo.flatMap((accountInfo: AccountInfo) => accountInfo.identities);
```

Comment:
This gives you a flat list of identities, which is not what we want. The structure looks like:

```
Account 0: [lisk identity, iov identity, eth identity]
Account 1: [lisk identity, iov identity, eth identity]
Account 2: [lisk identity, iov identity, eth identity]
Account 3: [lisk identity, iov identity, eth identity]
```

That outer list of accounts should be preserved. I think it is best to start the function with:

```ts
public async getBalances(): Promise<readonly (readonly Amount[])[]> {
  const accountsInfos = await this.accountManager.accounts();
  return accountsInfos.map((accountInfo: AccountInfo): readonly Amount[] => {
     // get all balances for accountInfo.identities
  });
}
```